include/libopencm3/stm32/h7/rcc.h

```diff
@@ -0,0 +1,458 @@
+#define RCC_CFGR_SWS_PLL1         0x3
+
+/* SW: System clock switch */
+#define RCC_CFGR_SW_SHIFT         0
+#define RCC_CFGR_SW_MASK          0x7
+#define RCC_CFGR_SW_HSI           0x0
+#define RCC_CFGR_SW_CSI           0x1
+#define RCC_CFGR_SW_HSE           0x2
+#define RCC_CFGR_SW_PLL1          0x3
+
+/**@}*/
+
+
+/* --- RCC_BDCR values ----------------------------------------------------- */
+#define RCC_BDCR_BDRST            (1 << 16)
+#define RCC_BDCR_RTCEN            (1 << 15)
+#define RCC_BDCR_RTCSEL_MASK      0x3
+#define RCC_BDCR_RTCSEL_SHIFT     8
+#define RCC_BDCR_RTCSEL_NONE      0
+#define RCC_BDCR_RTCSEL_LSE       1
+#define RCC_BDCR_RTCSEL_LSI       2
+#define RCC_BDCR_RTCSEL_HSE       3
+#define RCC_BDCR_LSEDRV_MASK      0x3
+#define RCC_BDCR_LSEDRV_SHIFT     3
+#define RCC_BDCR_LSEDRV_LOW       0
+#define RCC_BDCR_LSEDRV_MEDH      1 /* good job st */
+#define RCC_BDCR_LSEDRV_MEDL      2
+#define RCC_BDCR_LSEDRV_HIGH      3
+#define RCC_BDCR_LSEBYP           (1 << 2)
+#define RCC_BDCR_LSERDY           (1 << 1)
+#define RCC_BDCR_LSEON            (1 << 0)
+
+/* --- RCC_CSR values ------------------------------------------------------ */
+#define RCC_CSR_LSIRDY            (1 << 1)
+#define RCC_CSR_LSION             (1 << 0)
+
+extern uint32_t rcc_ahb_frequency;
+extern uint32_t rcc_apb1_frequency;
+extern uint32_t rcc_apb2_frequency;
+
+enum rcc_clock_3v3 {
+  RCC_CLOCK_3V3_216MHZ,
+  RCC_CLOCK_3V3_168MHZ,
+  RCC_CLOCK_3V3_120MHZ,
+  RCC_CLOCK_3V3_72MHZ,
+  RCC_CLOCK_3V3_48MHZ,
+  RCC_CLOCK_3V3_24MHZ,
+  RCC_CLOCK_3V3_END
+};
+
+// TODO: Customize this configuration based on the PLL options.
+struct rcc_clock_scale {
+  uint16_t plln;
+  uint8_t pllp;
+  uint8_t pllq;
+  uint32_t flash_waitstates;
+  uint8_t hpre;
+  uint8_t ppre1;
+  uint8_t ppre2;
+  enum pwr_svos_scale vos_scale;
+  uint8_t overdrive;
+  uint32_t ahb_frequency;
+  uint32_t apb1_frequency;
+  uint32_t apb2_frequency;
+};
+
+extern const struct rcc_clock_scale rcc_3v3[RCC_CLOCK_3V3_END];
+
+enum rcc_osc {
+  RCC_PLL,
+  RCC_HSE,
+  RCC_HSI,
+  RCC_LSE,
+  RCC_LSI
+};
+
+#define _REG_BIT(base, bit)       (((base) << 5) + (bit))
+
+enum rcc_periph_clken {
+   /* AHB1 peripherals */
+  RCC_DMA1        = _REG_BIT(0xD8, 0),
+  RCC_DMA2        = _REG_BIT(0xD8, 1),
+  RCC_ADC12       = _REG_BIT(0xD8, 5),
+  RCC_ETH1MAC     = _REG_BIT(0xD8, 15),
+  RCC_ETH1TX      = _REG_BIT(0xD8, 16),
+  RCC_ETH1RX      = _REG_BIT(0xD8, 17),
+
+  /* AHB2 peripherals */
+  RCC_DCMI        = _REG_BIT(0xDC, 0),
+  RCC_CRYP        = _REG_BIT(0xDC, 4),
+  RCC_HASH        = _REG_BIT(0xDC, 5),
+  RCC_RNG         = _REG_BIT(0xDC, 6),
+  RCC_SDMMC2      = _REG_BIT(0xDC, 9),
+  RCC_SRAM1       = _REG_BIT(0xDC, 29),
+  RCC_SRAM2       = _REG_BIT(0xDC, 30),
+  RCC_SRAM3       = _REG_BIT(0xDC, 31),
+
+  /* AHB3 peripherals */
+  RCC_MDMA        = _REG_BIT(0xD4, 0),
+  RCC_DMA2D       = _REG_BIT(0xD4, 4),
+  RCC_JPGDEC      = _REG_BIT(0xD4, 5),
+  RCC_FMC         = _REG_BIT(0xD4, 12),
```

Comment:
are you going to call it fmc or fmcc or fmsc?

Reply:
Everything here is setup identical to the F7.

Reply:
which, iirc, was setup based on f4, which was based on an early refman and not changed.  We may consider looking at them all again and getting one damn name for the thing, but I'll agree that it probably doesn't need to be done here then.